src/Microsoft.DotNet.Darc/DarcLib/AzureDevOpsClient.cs

```diff
@@ -1382,6 +1382,25 @@ await ExecuteAzureDevOpsAPIRequestAsync(
             baseAddressSubpath: "pkgs.");
     }
 
+    /// <summary>
+    ///   Fetches a list of last run AzDO builds for a given build definition.
+    /// </summary>
+    /// <param name="account">Azure DevOps account name</param>
+    /// <param name="project">Project name</param>
+    /// <param name="definitionId">Id of the pipeline (build definition)</param>
+    /// <param name="branch">Filter by branch</param>
+    /// <param name="count">Number of builds to retrieve</param>
+    /// <param name="status">Filter by status</param>
+    /// <returns>AzureDevOpsBuild</returns>
+    public async Task<JObject> GetBuildsAsync(string account, string project, int definitionId, string branch, int count, string status)
+        => await ExecuteAzureDevOpsAPIRequestAsync(
+            HttpMethod.Get,
+            account,
+            project,
+            $"_apis/build/builds?definitions={definitionId}&branchName={branch}&statusFilter={status}&$top={count}",
+            _logger,
+            versionOverride: "5.0");
```

Comment:
should this be some kind of cost in the class, or maybe in some configuration somewhere?

Reply:
I think we want to be flexible and set this per call

Reply:
Yeah, let's not re-write this whole class now